saleor/graphql/payment/mutations/transaction/utils.py

```diff
@@ -36,3 +41,25 @@ def get_transaction_item(id: str) -> payment_models.TransactionItem:
             }
         )
     return instance
+
+
+def clean_customer_ip_address(
+    info, customer_ip_address: Optional[str], error_code: str
+):
+    if not customer_ip_address:
+        return get_client_ip(info.context)
+    app = get_app_promise(info.context).get()
+    if not app or not app.has_perm(PaymentPermissions.HANDLE_PAYMENTS):
```

Comment:
Do you think it's worth to add comment here, that this field is protected? I'm not sure if I would get the idea behind that check if I didn't read the issue description.

Reply:
We require permission to set up the customerIpAddress manually via mutation input. We need this in case when app processes the payment instead of a customer.
I will add small comment here